### PR TITLE
Thymeleaf - Added extra convenience constructor

### PR DESCRIPTION
**Pull request by perwendel**
Thymeleaf - Added extra convenience constructor

### CODE REVIEW

**Comment by perwendel** [comment]
@tipsy please review (and perhaps test) :hamster:

File: spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java [hunk 1]
```diff
@@ -76,6 +89,26 @@ public String render(ModelAndView modelAndView) {
         } else {
             throw new IllegalArgumentException("modelAndView.getModel() must return a java.util.Map");
         }
+    }
 
+    private static TemplateResolver createDefaultTemplateResolver(String prefix, String suffix) {
+        TemplateResolver defaultTemplateResolver = new TemplateResolver();
+        defaultTemplateResolver.setTemplateMode(DEFAULT_TEMPLATE_MODE);
+
+        if (prefix != null) {
```
Comment (tipsy) [inline spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java]: How do you feel about: 
defaultTemplateResolver.setPrefix(
        prefix != null ? prefix : DEFAULT_PREFIX
);
        
defaultTemplateResolver.setSuffix(
        suffix != null ? suffix : DEFAULT_SUFFIX
);

File: spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java [hunk 2]
```diff
@@ -26,43 +26,56 @@
...
+     * @param templateResolver the template resolver.
      */
     public ThymeleafTemplateEngine(TemplateResolver templateResolver) {
+        initialize(templateResolver);
+    }
+
+    /**
+     * Initializes and sets the template resolver
+     */
+    private final void initialize(TemplateResolver templateResolver) {
```
Comment (tipsy) [inline spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java]: private final (final not needed)

File: spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java [hunk 3]
```diff
@@ -26,43 +26,56 @@
...
-
-        defaultTemplateResolver.setResourceResolver(new ClassLoaderResourceResolver());
+    /**
+     * Constructs a thymeleaf template engine with specified prefix and suffix
+     *
+     * @param prefix the prefix (template directory in resource path)
+     * @param suffix the suffix (e.g. .html)
+     */
+    public ThymeleafTemplateEngine(String prefix, String suffix) {
+        TemplateResolver defaultTemplateResolver = createDefaultTemplateResolver(prefix, suffix);
```
Comment (tipsy) [inline spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java]: Strange empty line

Comment (tipsy) [inline spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java]: Strange empty line

File: spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java [hunk 4]
```diff
@@ -26,43 +26,56 @@
...
 
-        templateEngine = new org.thymeleaf.TemplateEngine();
-        templateEngine.setTemplateResolver(defaultTemplateResolver);
+        initialize(defaultTemplateResolver);
```
Comment (tipsy) [inline spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java]: initialize(createDefaultTemplateResolver(prefix, suffix));

?

File: spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java (hunk 1 above)
Comment (perwendel) [inline spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java]: better

File: spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java (hunk 4 above)
Comment (perwendel) [inline spark-template-thymeleaf/src/main/java/spark/template/thymeleaf/ThymeleafTemplateEngine.java]: I think that makes it a bit harder to read. But it's a matter of taste.

**Pull request merged**